modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpAggregator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpRequest;
+
+import org.elasticsearch.http.HttpPreRequest;
+import org.elasticsearch.http.netty4.internal.HttpHeadersAuthenticatorUtils;
+
+import java.util.function.Predicate;
+
+public class Netty4HttpAggregator extends HttpObjectAggregator {
```

Comment:
basic question: considering the main point of `HttpObjectAggregator` is aggregating the incoming chucks, why are we extending it and essentially disabling its sole purpose by overriding `acceptInboundMessage`? Or is that not what is happening? if yes, shouldn't we just NOT use an aggregator?

Reply:
or maybe asked differently, is this (based on some predicate) how we enable this behaviour for endpoints that can handle streamed requests?

Reply:
My eventual goal is to have this be the _only_ way for endpoints to handle request bodies, leaving it up to the `RestHandler` implementation to decide how best to bound its memory usage. But that's likely a step too far for now.

Reply:
right, so any aggregation would be done on the rest handler, if it really needs to? (most likely this would be in some base class for them to use)

Reply:
> why are we extending it and essentially disabling its sole purpose by overriding acceptInboundMessage?

It's a bypass mechanism. Not accepted messages will flow further in pipeline, which enables streamed content handling. This code suppose to look at the request line and header of HttpRequest and make decision should it aggregate or pass through.

It might be possible to remove aggregator completely and feed chunks into JSON parser of rest request, rather than full body. For now we just allow optional bypass.